src/GitHub.Api/SimpleApiClient.cs

```diff
@@ -71,10 +71,10 @@ public class SimpleApiClient : ISimpleApiClient
                 }
             }
             // it'll throw if it's private or an enterprise instance requiring authentication
-            catch (ApiException)
+            catch (ApiException apiex)
             {
                 if (!HostAddress.IsGitHubDotComUri(OriginalUrl.ToRepositoryUrl()))
-                    isEnterprise = true;
+                    isEnterprise = apiex.HttpResponse?.Headers.ContainsKey("X-GitHub-Request-Id");
```

Comment:
I don't think so, desktop assumes that any exception means it's not an enterprise server, so if you're not authenticated, it isn't.
I'm being a bit more forceful and assuming that an exception might mean it's an authenticated enterprise server, which is why this causing doing false positives.